prometheus_client/openmetrics/exposition.py

```diff
@@ -67,24 +67,87 @@ def generate_latest(registry):
+                        s.native_histogram.zero_threshold,
+                        s.native_histogram.zero_count,
+                    ]
+
+                    # Signal presence for pos/neg spans/deltas
+                    pos = False
+                    neg = False
+
+                    # If there are pos spans, append them to the template and args
+                    if s.native_histogram.pos_spans:
+                        positive_spans = ','.join([f'{ps[0]}:{ps[1]}' for ps in s.native_histogram.pos_spans])
+                        positive_deltas = ','.join(f'{pd}' for pd in s.native_histogram.pos_deltas)
+                        nh_sample_template += ',positive_spans:[{}]'
+                        args.append(positive_spans)
+                        pos = True
+
+                    # If there are neg spans exist, append them to the template and args
+                    if s.native_histogram.neg_spans:
+                        negative_spans = ','.join([f'{ns[0]}:{ns[1]}' for ns in s.native_histogram.neg_spans])
+                        negative_deltas = ','.join(str(nd) for nd in s.native_histogram.neg_deltas)
+                        nh_sample_template += ',negative_spans:[{}]'
+                        args.append(negative_spans)
+                        neg = True
+
+                    # Append pos deltas if pos spans were added
+                    if pos:
+                        nh_sample_template += ',positive_deltas:[{}]'
+                        args.append(positive_deltas)
+
+                    # Append neg deltas if neg spans were added
+                    if neg:
+                        nh_sample_template += ',negative_deltas:[{}]'
+                        args.append(negative_deltas)
+
+                    # Add closing brace
+                    nh_sample_template += '}}'
+
+                    # Format the template with the args
+                    native_histogram = nh_sample_template.format(*args)    
+
+                value = ''    
+                if s.value is not None or not s.native_histogram:
+                    value = floatToGoString(s.value)
                 if _is_valid_legacy_metric_name(s.name):
-                    output.append('{}{} {}{}{}\n'.format(
+                    output.append('{}{} {}{}{}{}\n'.format(
                         s.name,
                         labelstr,
-                        floatToGoString(s.value),
+                        value,
                         timestamp,
                         exemplarstr,
+                        native_histogram
                     ))
                 else:
-                    output.append('{} {}{}{}\n'.format(
+                    output.append('{} {}{}{}{}\n'.format(
                         labelstr,
-                        floatToGoString(s.value),
+                        value,
                         timestamp,
                         exemplarstr,
+                        native_histogram
```

Comment:
this must be placed immediately after `value,` because the timestamp and exemplars come last